### PR TITLE
Target tensorflow stable instead of tensorflow-nightly.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
Linked to #893

I'll try to list here the pros and cons so that we can decide if we go forward with this or not:

Pros:

* The CI can't fail out of nowhere because or a bug/API change on the Tensorflow side. The CI is more deterministic than before.
* The bump of the tensorflow version can be done in a isolated manner, in a pull request.
* We can make a minimal Dockerfile used for testing and tooling locally, and include bazel and tensorflow inside. Contributors won't have to download tf-nightly every day. Build once, run every day :)
* We don't have to support a single version of tensorflow, we can support a range of them and test against the oldest and the most recent in the CI: (eg: >=2.0.0,<2.2.0)
* If the dependency doesn't change every day, it means that the unit tests are reproductible. We can then use tools like `git biscect` to detect the commit which introduced a bug in the master branch.

Cons:

* We can't use the latest features available in tf-nightly.
* We have to bump the version number manually because it will be pinned, before it was done automatically.
* People using TensorFlow nightly might not be able to use the latest version of the addons master branch if TensorFlow inserted a regression or an API change.

To be honest, all the points made above (pros and cons) are pretty weak because tensorflow have strong API garantees. Those are generalities about pinning the version numbers in CIs. 

The only strong points might be the ones about tensorflow bugs and about not having to download tf-nightly every day when using Docker locally to test things.

### CODE REVIEW

**Comment by facaiy** [comment]
@seanpmorgan @tensorflow/sig-addons-maintainers what's your opinion?

**Comment by seanpmorgan** [comment]
Seeing as there should be no regressions inserted into TF (barring experimental features which we do occasionally use), I'd be in favor of our unit tests being built against a stable TF release. 

As a check against drifting though, I would like to setup a periodic build/test against tf-nightly though so come release time we don't suddenly find out about several breaking issues.

Does that seem agreeable?

File: requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-tf-nightly
\ No newline at end of file
+tensorflow==2.1.0
```
Comment (seanpmorgan) [inline requirements.txt]: How would we handle this if #864 goes through? Make this file into `requirements-test.txt`?

**Comment by gabrieldemarmiesse** [comment]
> As a check against drifting though, I would like to setup a periodic build/test against tf-nightly though so come release time we don't suddenly find out about several breaking issues.

Isn't it the primary purpose of release candidates? We'll have several weeks to adapt before the final release :)

**Comment by seanpmorgan** [comment]
> > As a check against drifting though, I would like to setup a periodic build/test against tf-nightly though so come release time we don't suddenly find out about several breaking issues.
> 
> Isn't it the primary purpose of release candidates? We'll have several weeks to adapt before the final release :)

Fair. I guess I'm just still scarred from trying to keep up with breaking changes during TF2 beta. That should be sufficient and if we find out that it's not we can re-look into how we deal with that.

**Comment by gabrieldemarmiesse** [comment]
for example, we can say that we're compatible with tf 2.1.0-2.2.0 and test against that. Then, when the release candidate for 2.3.0 comes, someone can make a PR to bump the version number in the CI. The CI will then run on 2.1.0 and 2.3.0rc0

Then on 2.1.0 and 2.3.0rc1 
then on 2.1.0 and 2.3.0.

I believe it's simpler than to setup periodic builds :) it's also easier for open source beginners to understand :) :)

**Comment by seanpmorgan** [comment]
@gabrieldemarmiesse small conflict since merging the typegaurd PR

**Pull request merged**